core/docker/build.sh

```diff
@@ -66,18 +71,38 @@ tar -C "${WORK_DIR}" -xzf "${WORK_DIR}/trino-server-${TRINO_VERSION}.tar.gz"
 rm "${WORK_DIR}/trino-server-${TRINO_VERSION}.tar.gz"
 cp -R bin "${WORK_DIR}/trino-server-${TRINO_VERSION}"
 cp -R default "${WORK_DIR}/"
+BASE_WORK_DIR="$(mktemp -d)"
 
+BASE_PREFIX="trino-base:${TRINO_VERSION}"
 TAG_PREFIX="trino:${TRINO_VERSION}"
 
+echo "👷 Docker buildx builders available:"
+docker buildx ls
+
 for arch in "${ARCHITECTURES[@]}"; do
-    echo "🫙  Building the image for $arch"
-    docker build \
-        "${WORK_DIR}" \
-        --pull \
-        --platform "linux/$arch" \
-        -f Dockerfile \
-        -t "${TAG_PREFIX}-$arch" \
-        --build-arg "TRINO_VERSION=${TRINO_VERSION}"
+    echo "🫙  Building the base image for $arch"
+    BASE_BUILD_OPTIONS=(
+        ${BASE_WORK_DIR}
+        --platform "linux/$arch"
+        --load
+        --tag "${BASE_PREFIX}-$arch"
+        -f trino-base.Dockerfile
+        )
+    if [ -n "${BUILD_CACHE}" ]; then
+        echo "🎯 Using Docker build cache"
+        BASE_BUILD_OPTIONS+=(--cache-from "${BUILD_CACHE},scope=${GITHUB_REF_NAME}-${arch}" --cache-to "${BUILD_CACHE},scope=${GITHUB_REF_NAME}-${arch},mode=max")
```

Comment:
If `GITHUB_REF_NAME` is not set, maybe fall back to the output of `git rev-parse --short HEAD`?